src/input.h

```diff
@@ -634,6 +634,10 @@ class input_context
          */
         input_event get_raw_input();
 
+        /**
+         * Get coordinate of text level from mouse input, difference between this and get_coordinates is that one is getting pixel level coordinate.
+         * Return false if the result is not in the window.
+         */
         std::pair<point, bool> get_coordinates_text( const catacurses::window &capture_win ) const;
```

Comment:
```suggestion
        cata::optional<point> get_coordinates_text( const catacurses::window &capture_win ) const;
```
Currently this returns a point **and** a boolean. That means it returns a point, even if the point is not within the window.

This would also make its usage simpler:
```C++
if( const cata::optional<point> p = ctxt.get_coordinates_text( w_pickup ) ) {
    print( ..., *p, ... );
}
```

Reply:
that's reserved for future usage in case of need the point out side of the window

Reply:
okey, i decided to seperate checking of is in window from it